lib/ecto/adapters/postgres.ex

```diff
@@ -344,10 +344,19 @@ defmodule Ecto.Adapters.Postgres do
   end
 
   defp select_versions(table, config) do
-    case run_query(~s[SELECT version FROM public."#{table}" ORDER BY version], config) do
-      {:ok, %{rows: rows}} -> {:ok, Enum.map(rows, &hd/1)}
-      {:error, %{postgres: %{code: :undefined_table}}} -> {:ok, []}
+    migration_prefixes = config[:migration_prefixes] || ["public"]
+
+    result = Enum.reduce_while(migration_prefixes, [], fn prefix, prefix_versions ->
+      case run_query(~s[SELECT version FROM #{prefix}."#{table}" ORDER BY version], config) do
+        {:ok, %{rows: rows}} -> {:cont, prefix_versions ++ Enum.map(rows, &{prefix, hd(&1)})}
+        {:error, %{postgres: %{code: :undefined_table}}} -> {:cont, prefix_versions}
+        {:error, _} = error -> {:halt, error}
+      end
+    end)
+
+    case result do
       {:error, _} = error -> error
+      versions -> {:ok, versions}
```

Comment:
I think we should be consistent. Some places we are calling it `prefix_versions` and most places `versions`.  Probably we can stick with `versions` everywhere?